client.go

```diff
@@ -0,0 +1,727 @@
+package redis
+
+import (
+	"time"
+
+	"github.com/dop251/goja"
+	"github.com/go-redis/redis/v8"
+	"go.k6.io/k6/js/modules"
+)
+
+// Client represents the Client constructor (i.e. `new redis.Client()`) and
+// returns a new Redis client object.
+type Client struct {
+	vu          modules.VU
+	redisClient *redis.Client
+}
+
+// Set the given key with the given value.
+func (c *Client) Set(key string, value interface{}) *goja.Promise {
```

Comment:
Using `interface{}` directly IMO lends this to a race. Using `string`, `int` and other not  pointer (string is immutable) types.

While I would expect that sending anything that can be mutated from the goja to `redisCleint.Set` it still can lead to a race.

This might be less relevant here, but is likely more relevant in all the other places where you take `interface{}` or even more importantly`...interface{}`. 


I would argue the type of `value` should be checked to be one of few (int. string, float ?!?), and return an error without even going in the goroutine, if not.

I also noticed that you set the expiration to 0, but there is no problem to just have a third argument, and it will be 0 populated by goja if not value is provided ;).

Reply:
Those are excellent points! Thanks a lot for your input, much appreciated! Going to add those checks indeed, and that timeout argument 🙇🏻